File: server/models/AINews.js
// models/AINews.js
const mongoose = require('mongoose');

const aiNewsSchema = new mongoose.Schema({
  title: { type: String, required: true },
  excerpt: String,
  link: String,
  date: { type: Date, default: Date.now },
  source: String,
  author: String,
  image: String,
}, { timestamps: true });

module.exports = mongoose.model('AINews', aiNewsSchema);

File: server/models/ai_news.py
import datetime
from mongoengine import Document, StringField, DateTimeField

class AINews(Document):
    title = StringField(required=True)
    excerpt = StringField()
    link = StringField()
    date = DateTimeField(default=datetime.datetime.utcnow)
    source = StringField()
    author = StringField()
    image = StringField()

    meta = {'collection': 'ai_news'}
